test/hash_test.rb

```diff
@@ -33,29 +33,40 @@
       hash[:fub].must_equal('baz')
     end
 
-    it 'symbolize nested hashes' do
-      hash = Hanami::Utils::Hash.new('nested' => { 'key' => 'value' })
-      hash.symbolize!
+    describe 'with deep true' do
+      it 'symbolize nested hashes' do
+        hash = Hanami::Utils::Hash.new('nested' => { 'key' => 'value' })
+        hash.symbolize!
 
-      hash[:nested].must_be_kind_of Hanami::Utils::Hash
-      hash[:nested][:key].must_equal('value')
-    end
+        hash[:nested].must_be_kind_of Hanami::Utils::Hash
+        hash[:nested][:key].must_equal('value')
+      end
 
-    it 'symbolize nested Hanami::Utils::Hashes' do
-      nested = Hanami::Utils::Hash.new('key' => 'value')
-      hash = Hanami::Utils::Hash.new('nested' => nested)
-      hash.symbolize!
+      it 'symbolize nested Hanami::Utils::Hashes' do
+        nested = Hanami::Utils::Hash.new('key' => 'value')
+        hash = Hanami::Utils::Hash.new('nested' => nested)
+        hash.symbolize!
 
-      hash[:nested].must_be_kind_of Hanami::Utils::Hash
-      hash[:nested][:key].must_equal('value')
+        hash[:nested].must_be_kind_of Hanami::Utils::Hash
+        hash[:nested][:key].must_equal('value')
+      end
+
+      it 'symbolize nested object that responds to to_hash' do
+        nested = Hanami::Utils::Hash.new('metadata' => WrappingHash.new('coverage' => 100))
+        nested.symbolize!
+
+        nested[:metadata].must_be_kind_of Hanami::Utils::Hash
+        nested[:metadata][:coverage].must_equal(100)
+      end
     end
 
-    it 'symbolize nested object that responds to to_hash' do
-      nested = Hanami::Utils::Hash.new('metadata' => WrappingHash.new('coverage' => 100))
-      nested.symbolize!
+    describe 'with deep false' do
+      it 'does not simbolize nested hashes' do
+        hash = Hanami::Utils::Hash.new('nested' => { 'key' => 'value' })
+        hash.symbolize!(deep: false)
 
-      nested[:metadata].must_be_kind_of Hanami::Utils::Hash
-      nested[:metadata][:coverage].must_equal(100)
+        hash[:nested]['key'].must_equal('value')
```

Comment:
What do you think about checking the `keys` of `hash[:nested]` instead? To me it is clearer in intent as we do only assert that the key is not symoblized